src/refiners/fluxion/layers/attentions.py

```diff
@@ -37,11 +39,24 @@ def sparse_dot_product_attention_non_optimized(
 
 
 class ScaledDotProductAttention(Module):
-    def __init__(self, num_heads: int = 1, is_causal: bool | None = None, is_optimized: bool = True) -> None:
+    def __init__(
+        self,
+        num_heads: int = 1,
+        is_causal: bool | None = None,
+        is_optimized: bool = True,
+        slice_size: Optional[int] = 4096,
+    ) -> None:
         super().__init__()
         self.num_heads = num_heads
         self.is_causal = is_causal
         self.is_optimized = is_optimized
+        # 4096 max needed for SD 1.5 512x512
+        # 9216 max needed for SD 1.5 768x768
+        # 16384 max needed for SD 1.5 1024x1024
+        # 32400 max needed for SD 1.5 1920x1080 (HD)
+        # 129600 max needed for SD 1.5 3840x2160 (4k)
+        # 234375 max needed for SD 1.5 5000x3000
```

Comment:
`ScaledDotProductAttention` shouldn't "know" that SD exists, maybe you can document this in the `SD1UNet` docstring

Reply:
True, and I wasn't sure where this should be documented, I would have added some relevant SDXL values as well if I had that implemented. I just used it to inform the default I picked.  Also feels weird for the unet to have this comment.  I'd be okay just deleting it.  Perhaps it fits best in the `_sliced_attention` function.

Reply:
I made the comment just explain the default and moved it to `_sliced_attention`

Reply:
For this [exact same reason](https://github.com/finegrain-ai/refiners/pull/130#discussion_r1412100335), I think you should either delete this comment or move it at a higher-level - UNet/SD - depending on what's decided to act on this setting

Reply:
okay I will delete this comment even though I think it adds value :-) 
<img width="725" alt="image" src="https://github.com/finegrain-ai/refiners/assets/1217531/d86e9766-6bd6-4a59-98db-8317b86caa1e">

Reply:
comment removed